lib/lwm2m_carrier/include/lwm2m_carrier.h

```diff
@@ -49,6 +50,49 @@ typedef struct {
 	size_t psk_length;
 } lwm2m_carrier_config_t;
 
+/**
+ * @brief LWM2M device power sources type.
+ */
+typedef enum {
+    LWM2M_CARRIER_POWER_SOURCE_DC                        = 0,
+    LWM2M_CARRIER_POWER_SOURCE_INTERNAL_BATTERY          = 1,
+    LWM2M_CARRIER_POWER_SOURCE_EXTERNAL_BATTERY          = 2,
+    LWM2M_CARRIER_POWER_SOURCE_ETHERNET                  = 4,
+    LWM2M_CARRIER_POWER_SOURCE_USB                       = 5,
+    LWM2M_CARRIER_POWER_SOURCE_AC                        = 6,
+    LWM2M_CARRIER_POWER_SOURCE_SOLAR                     = 7
+} lwm2m_carrier_power_source_t;
```

Comment:
I don't like the use of enums in an API of a precompiled library. This is error-prone, as it enforces the exact same enum configuration in the compiler, as the library was built with (see `-fshort-enums`). If an application use a different setup, it will likely end up in a hard-to-investigate crash.

Reply:
TIL https://oroboro.com/short-enum/